File: app/src/main/java/com/akilimo/mobile/inherit/BaseDialogFragment.java
package com.akilimo.mobile.inherit;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import com.akilimo.mobile.dao.AppDatabase;
import com.akilimo.mobile.utils.MathHelper;
import com.akilimo.mobile.utils.SessionManager;

@SuppressWarnings("WeakerAccess")
public abstract class BaseDialogFragment extends DialogFragment {

    protected String LOG_TAG = BaseDialogFragment.class.getSimpleName();


    protected MathHelper mathHelper;
    protected Context context;
    protected AppDatabase database;
    protected SessionManager sessionManager;
    protected String currencySymbol;


    public BaseDialogFragment() {
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        mathHelper = new MathHelper();
        database = AppDatabase.getDatabase(context);
    }
}

File: app/src/main/java/com/akilimo/mobile/inherit/BaseDialogFragment.kt
package com.akilimo.mobile.inherit

import androidx.fragment.app.DialogFragment
import com.akilimo.mobile.dao.AppDatabase
import com.akilimo.mobile.utils.MathHelper
import com.akilimo.mobile.utils.PreferenceManager
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

@Deprecated(
    "Use BaseBottomSheetDialogFragment instead",
    replaceWith = ReplaceWith("BaseBottomSheetDialogFragment")
)
abstract class BaseDialogFragment : DialogFragment() {
    protected val mathHelper: MathHelper by lazy { MathHelper() }
    protected val sessionManager: PreferenceManager by lazy { PreferenceManager(requireContext().applicationContext) }
    protected val database: AppDatabase by lazy { AppDatabase.getInstance(requireContext().applicationContext) }

    protected var currencySymbol: String = "USD"
}

abstract class BaseBottomSheetDialogFragment : BottomSheetDialogFragment() {
    protected val mathHelper: MathHelper by lazy { MathHelper() }
    protected val sessionManager: PreferenceManager by lazy { PreferenceManager(requireContext().applicationContext) }
    protected val database: AppDatabase by lazy { AppDatabase.getInstance(requireContext().applicationContext) }
    protected var currencySymbol: String = "USD"
}